packages/api/src/utils/logParser.ts

```diff
@@ -233,13 +251,13 @@ abstract class ParsingInterface<T> {
   abstract _parse(
     log: T,
     ...args: any[]
-  ): LogStreamModel | MetricModel | RrwebEventModel;
+  ): Promise<LogStreamModel | MetricModel | RrwebEventModel>;
 
-  parse(logs: T[], ...args: any[]) {
+  async parse(logs: T[], ...args: any[]) {
     const parsedLogs: any[] = [];
     for (const log of logs) {
       try {
-        parsedLogs.push(this._parse(log, ...args));
+        parsedLogs.push(await this._parse(log, ...args));
```

Comment:
I believe this was the line I missed. It was not awaited in the old version:

https://github.com/hyperdxio/hyperdx/blob/318f5d82f95bb1153a531fcb55d2596d1fad1cef/packages/api/src/utils/logParser.ts#L253